lib/puppet/pops/loader/module_loaders.rb

```diff
@@ -19,6 +19,18 @@ module Loader
 # @api private
 #
 module ModuleLoaders
+  # This is exactly the same as the #system_loader_from method, but the argument for path is changed to
+  # location where pluginsync stores functions.
+  def self.cached_loader_from(parent_loader, loaders)
+    LibRootedFileBased.new(parent_loader,
+      loaders,
+      nil,
+      Puppet[:libdir],
+      'cached_puppet_lib',
+      [:func_4x, :datatype]
```

Comment:
Should this be restricted to only load the :func_4x initially? Or should we also just load the :datatypes as well?